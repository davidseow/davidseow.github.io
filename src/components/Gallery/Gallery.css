img {
  box-sizing: border-box;
  display: block;
  padding: 3px;
  width: 100%;
}
