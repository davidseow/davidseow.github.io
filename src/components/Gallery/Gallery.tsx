import React from "react";
import Masonry, { ResponsiveMasonry } from "react-responsive-masonry";

import "./Gallery.css";

const Gallery = (props) => {
  const { images } = props;

  return (
    <ResponsiveMasonry
      columnsCountBreakPoints={{ 350: 1, 750: 2, 900: 3 }}
      className="galleryContainer"
    >
      <Masonry className="gallery">
        {images.map((image: string, i: Number) => (
          <img key={i.toString()} src={image} alt="" />
        ))}
      </Masonry>
    </ResponsiveMasonry>
  );
};
export default Gallery;
