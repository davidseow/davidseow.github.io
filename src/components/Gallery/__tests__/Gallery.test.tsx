import React from "react";
import { render } from "@testing-library/react";
import Gallery from "../Gallery";

describe("Gallery", () => {
  it("should render", () => {
    const images = [
      "/photos/DSC01435.jpg",
      "/photos/DSCF0469.jpg",
      "/photos/DSCF1140.jpg",
      "/photos/DSCF1264.jpg",
      "/photos/DSCF3189.jpg",
      "/photos/DSCF3526.jpg",
      "/photos/DSCF5285.jpg",
    ];
    const { asFragment } = render(<Gallery images={images} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
