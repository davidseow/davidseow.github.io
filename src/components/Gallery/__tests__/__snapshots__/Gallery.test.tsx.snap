// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render 1`] = `
<DocumentFragment>
  <div
    class="galleryContainer"
  >
    <div
      class="gallery"
      style="display: flex; flex-direction: row; justify-content: center; align-content: stretch; box-sizing: border-box; width: 100%; gap: 0;"
    >
      <div
        style="display: flex; flex-direction: column; justify-content: flex-start; align-content: stretch; flex: 1; width: 0px; gap: 0;"
      >
        <img
          alt=""
          src="/photos/DSC01435.jpg"
        />
        <img
          alt=""
          src="/photos/DSCF1264.jpg"
        />
        <img
          alt=""
          src="/photos/DSCF5285.jpg"
        />
      </div>
      <div
        style="display: flex; flex-direction: column; justify-content: flex-start; align-content: stretch; flex: 1; width: 0px; gap: 0;"
      >
        <img
          alt=""
          src="/photos/DSCF0469.jpg"
        />
        <img
          alt=""
          src="/photos/DSCF3189.jpg"
        />
      </div>
      <div
        style="display: flex; flex-direction: column; justify-content: flex-start; align-content: stretch; flex: 1; width: 0px; gap: 0;"
      >
        <img
          alt=""
          src="/photos/DSCF1140.jpg"
        />
        <img
          alt=""
          src="/photos/DSCF3526.jpg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
